crates/context/src/lib.rs

```diff
@@ -332,10 +352,13 @@ impl ContextManager {
 
         let context_exists = handle.has(&ContextMetaKey::new(context_id))?;
 
+        let proxy_contract = self.get_proxy_contract(context_id).await;
+
         let mut config = (!context_exists).then(|| ContextConfigParams {
```

Comment:
I'd move this into the block, since we only need to call it when the context doesn't exist

this will require a bit of a refactor